File: DemoMMOUnity/Assets/Game/Net/Script/Net/CsProtocol/Protocol/Login/GetPlayerInfoRequest.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Spring.Utils;

namespace ZJYFrameWork.Net.CsProtocol.Buffer
{
    public class GetPlayerInfoRequest : Model, IPacket
    {
        public string token;

        public static GetPlayerInfoRequest ValueOf(string token)
        {
            var packet = new GetPlayerInfoRequest
            {
                token = token
            };
            return packet;
        }

        public short ProtocolId()
        {
            return 1004;
        }
    }

    public class GetPlayerInfoRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1004;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                buffer.WriteBool(false);
                return;
            }

            var infoRequest = (GetPlayerInfoRequest)packet;
            var message = new ServerMessageWrite(infoRequest.ProtocolId(), infoRequest);
            var json = JsonConvert.SerializeObject(message);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            return null;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/login/getPlayerInfoRequest.lua
---@class GetPlayerInfoRequest
local GetPlayerInfoRequest = class("GetPlayerInfoRequest")
function GetPlayerInfoRequest:ctor()
    --- 返回加密的信息
    ---@type string
    self.token = string.empty
end

---@param token string 返回加密的信息
---@return GetPlayerInfoRequest
function GetPlayerInfoRequest:new(token)
    self.token = token --- java.lang.String
    return self
end

---@return number
function GetPlayerInfoRequest:protocolId()
    return 1004
end

---@return string
function GetPlayerInfoRequest:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            token = self.token
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return GetPlayerInfoRequest
function GetPlayerInfoRequest:read(data)

    local packet = self:new(
            data.token)
    return packet
end

--- 返回加密的信息
---@type  string 返回加密的信息
function GetPlayerInfoRequest:getToken()
    return self.token
end



return GetPlayerInfoRequest
